Bottom: e7f3d3be78e2f7bfbc98b68ca7540794ea2ebca7
Top:    bd7376f4507bf3b6043c2fe186b84ef48bfdc118
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-09-09 10:43:17 +0100

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index a8dc5c0..194d279 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1163,7 +1163,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
